.github/workflows/submitty_ci.yml

```diff
@@ -381,22 +381,61 @@ jobs:
             run: |
               sudo systemctl restart submitty_autograding_worker
               sudo systemctl restart submitty_autograding_shipper
+              export DISPLAY=:99
               chromedriver --url-base=/wd/hub &
-          - name: Run e2e tests with Pam auth
+
+          - name: Run selenium e2e tests with pam auth
             run: |
               pushd ${SUBMITTY_REPOSITORY}
               echo "Authentication Method => $(sudo jq -r ".authentication_method" /usr/local/submitty/config/database.json)"
               TEST_URL="http://localhost" python3 -m unittest discover -v --start-directory tests
               sudo bash tests/test_site_error_log.sh
-          - name: Run login tests with database auth
+
+          - name: Cache Node Modules
+            uses: actions/cache@v2
+            with:
+              path: ~/.npm
+              key: ${{ runner.os }}-cache-node-modules-dev-${{ hashFiles('**/package-lock.json') }}
+              restore-keys: |
+                 ${{ runner.os }}-cache-node-modules-dev
+          - name: Install Cypress
```

Comment:
So for some reason cypress is trying to pull in a javascript file and fails to do so:
here is the error log
```
/usr/local/bin/npm ci
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/eligrey/FileSaver.js.git
npm ERR! 
npm ERR! Warning: Permanently added the RSA host key for IP address '140.82.113.3' to the list of known hosts.
npm ERR! git@github.com: Permission denied (publickey).
npm ERR! fatal: Could not read from remote repository.
npm ERR! 
npm ERR! Please make sure you have the correct access rights
npm ERR! and the repository exists.
npm ERR! 
npm ERR! exited with error code: 128

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2021-03-04T01_21_35_660Z-debug.log
Error: The process '/usr/local/bin/npm' failed with exit code 1
```

just tested now in this run : https://github.com/Submitty/Submitty/runs/2027066212?check_suite_focus=true

I'm not sure why this is but manually installing the `file-saver` package fixes it, which is why there's that specific npm install step for cypress.

Reply:
This is because `jsPDF` has a dependency on `file-saver` but uses a direct link to GH repo instead of a NPM release. The easiest solution is probably to first upgrade to the newest version of `jsPDF` which goes back to using npm releases for `file-saver`, and then we can drop a whole bunch of lines here.

Reply:
Ok I will leave this alone for now